util/pace/util/_profiler.py

```diff
@@ -0,0 +1,44 @@
+import cProfile
+
+
+class Profiler:
+    def __init__(self):
+        self._enabled = True
+        self.profiler = cProfile.Profile()
+        self.profiler.disable()
+
+    def enable(self):
+        self.profiler.enable()
+
+    def dump_stats(self, filename: str):
+        self.profiler.disable()
+        self._enabled = False
+        self.profiler.dump_stats(filename)
+
+    @property
+    def enabled(self) -> bool:
+        """Indicates whether the profiler is currently enabled."""
+        return self._enabled
+
+
+class NullProfiler(Profiler):
```

Comment:
why are we inheriting from the base-profiler here? This seems dangerous. If we want a base profiler it should be an abc, right?

Reply:
Removed inheritance